### DISCUSSION TITLE
Specify custom sorting on split-by/group-by entries in datagrid

### DISCUSSION THREAD

**Discussion by ja-b** [perspective-dev/perspective#2002] [Q&A]
## Support Question

How do I specify custom sorting on split-by/group-by entries in the perspective-viewer-datagrid?

Say I have the following

![image](https://user-images.githubusercontent.com/931645/199095556-a774dad0-f5f7-4de9-9572-b6988f754081.png)

Where I'd like to sort the `18M`, `1Y`, `2Y` etc. columns differently than the default string-sorting method built-in (one thats aware of the spans they represent). I tried looking for an example of this but couldn't find anything.

I'm interacting with this via Panel, which uses the Perspective Python-API under the hood.

**Comment by texodus** [comment]
You need a column sort e.g. `"col desc"` or the 3rd/4th click state for the sort icon, as well as a category aggregate e.g. `dominant` or `unique`.  You can provide this sort order in your data set (e.g. `tenor sort order`) or you can calculate it in an expression column using the `order()` function, it would look something like this:
```javascript
viewer.restore({ 
    sort: [["tenor_sort", "col desc"], ...],
    aggregate: {"tenor_sort": "dominant" ...},
    expressions: [`//tenor_sort\norder("Tenor", '6m', '18m', .... )`, ...],
   ....
})
```

... or as an expression in the UI (note in both examples `"` vs `'` for column names vs string values)

```
order("tenor", '6m', '18m', ... )
```
